### DISCUSSION TITLE
'SET_PRINT_STATS_INFO' not found in gcode_macro rename

### DISCUSSION THREAD

**Discussion by leggler** [jschuh/klipper-macros#188] [Q&A] [answered]
Hello,
I tried Installing your klipper-macro set, following the install-instructions given in this repo.

After removing all old macros existing before in my config, and changing the prusaslicer settings accordingly, I still endet up with error:

`Existing command 'SET_PRINT_STATS_INFO' not found in gcode_macro rename`

I guess it has something to do with the layers.cfg macro, as this seems to be the only one using "PRINT_STATS_INFO"... but I am not expert engough to make any changes there...

Btw, i do NOT use mainsail-macros (because this error seems to also have happended using their macro configuration)

**Comment by leggler** [comment]
I can confirm that it was related to layers.config - removing it from the imports did the trick (with loosing the functionality of course).

I now run into problems with 
`Option 'on_error_gcode' is not valid in section 'virtual_sdcard'`

my setting is exactly the one in the install-instructions (path being correct already):
[virtual_sdcard]
path: ~/gcode_files
on_error_gcode: CANCEL_PRINT

PS: I understand that globals.cfg makes it a requirement to defien an on_error_gcode. I wonder why this is required - I didnt have one before and it worked fine, and the most dangerous thing that could go wrong is a failed print...

PPS: I also tried to define a simple CANCLE_PRINT makro directly in printer.cfg, but the error message remained the same.

**Reply by leggler** [reply]
I tried changing globals.cfg, removing the on_error_gcode check, but when I get one problem out of the way another apears.

Will change back to my old macros now, its a pitty, would have liked to try out the partial probing feature

**Comment by jschuh** [comment, marked as answer]
The SET_PRINT_STATS_INFO gcode has been supported in Klipper for more than a year. So, you hit this error because you're running an old, unsupported version of Klipper.

**Comment by leggler** [comment]
Thanks for the Hint! I suppose you are refering to the software running on my pi, not the firmware version of the controller board?

I am running the latest version of the octoprint klipper plugin available (https://github.com/thelastWallE/OctoprintKlipperPlugin). but I see now, that this is arleady from febraury 23... Seems like octoprint has an outdated klipper version in the back... do you have a recommondation for which interface to use?

![grafik](https://github.com/jschuh/klipper-macros/assets/16446299/26441735-bb3a-48a4-ae0c-b4b69fe60169)

**Comment by leggler** [comment]
after updating Klipper (Needed to do a git-pull in the klipper folder of the raspberry) it worked.
I summarize the steps to update Klipper on octoprint here, as this took me some time to figure out. 
This can not be done from the octoprint interface, but needs you to connect ot the pi via ssh.

updating steps
**1. SSH into Your Raspberry Pi:**
        Use an SSH client like PuTTY (for Windows) or the terminal (for macOS and Linux) to connect to your Raspberry Pi.
        The command usually is s`sh pi@<your-raspberry-pi-ip-address>`. Replace `<your-raspberry-pi-ip-address>` with the actual IP address of your Raspberry Pi. The default username is pi and the default password is raspberry, unless you've changed them.

**2. Navigate to the Klipper Directory:**

Once logged in, navigate to the Klipper directory using the command: `cd ~/klipper`

**3.   Stop the Klipper Service:**

Before updating, you need to stop the Klipper service. Use the command: `sudo service klipper stop`

**4.   Fetch the Latest Code:**

Now, update the Klipper software to the latest version with the command: `git pull`

**5.   Recompile the Klipper Firmware:**

Recompile the firmware with `make clean` followed by `make`.
This step is necessary to ensure that the firmware running on your printer's control board is updated along with the software on your Raspberry Pi.

**6.   Flash the Firmware to Your Printer (if needed):**

(Using BIGTREETECH SKR Mini E3 V2.0, that I already had klipper running on, id didnt need to)

**7.    Restart the Klipper Service:**

Once everything is updated and the new firmware is flashed (if needed), restart the Klipper service with: `sudo service klipper start`

**8.    Check OctoPrint:**

Finally, go to the OctoPrint interface in your web browser and check if everything is running smoothly. You might need to reconnect OctoPrint to your printer.

**Comment by rpgiacon** [comment]
Use SSH  go to the mainsail-config folder "sudo chmod 765 client.cfg" then use nano (sudo nano client.cfg) and comment the line rename_existing: SET_PRINT_STATS_INFO_BASE (#rename_existing: SET_PRINT_STATS_INFO_BASE). save and restart
